File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.py
class Solution(object):
    def longestMonotonicSubarray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        result=1
        s_dec=1
        s_inc=1

        for i in range(1,len(nums)):
            if(nums[i]>nums[i-1]):
                s_inc+=1
                s_dec=1
            elif (nums[i]<nums[i-1]):
                s_dec+=1
                s_inc=1
            else:
                s_dec=1
                s_inc=1
            result=max(result, max(s_dec,s_inc))
        return result
        

File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.cpp
class Solution {
public:
    int longestMonotonicSubarray(vector<int>& nums) {
        int result=1,s_inc=1,s_dec=1;

        for(int i=1;i<nums.size();i++){

            if(nums[i-1]>nums[i]){
                s_dec++;
                s_inc=1;
            }else if(nums[i-1]<nums[i]){
                s_inc++;
                s_dec=1;
            }else{
                s_inc=1;
                s_dec=1;
            }
            result=max(result,max(s_inc,s_dec));
        }
        return result;
    }
};